packages/ripple/src/getRippleGlobalNamespace.ts

```diff
@@ -0,0 +1,37 @@
+interface ModuleType {
+  '@leafygreen-ui/ripple': {
+    // @ts-expect-error using HTMLElements to index as it provides a faster lookup when deciding if we should create a ripple effect on a given element
+    registeredRippleElements: { [K: HTMLElement]: {} };
+    setRippleListener: boolean;
+  };
+}
+
+type LGWindow = Window &
+  typeof globalThis & {
+    __LEAFYGREEN_UTILS__: {
+      modules: ModuleType;
+    };
+  };
+
+export function getRippleGlobalNamespace() {
+  if (!window) {
```

Comment:
Are we worrying about server side rendering at this point?

Reply:
It'd be nice to have our own internal `globalThis` shim that we could rely on across all packages.

Reply:
updated to check for if window is not undefined, because we are worrying about ssr

Reply:
Also: The reason I didn't make this more global and include it in lib was because we didn't want React to be a dep of this package (and React is a dep of lib). I can make another external package that handles interacting with our global namespace in a more abstract/reusable way, but figured I'd do that once we needed it again. What are your thoughts?

Reply:
Agreed on all thoughts!

Reply:
SGTM